src/test/java/programminglife/model/drawing/SubGraphSNPTest.java

```diff
@@ -108,9 +105,6 @@ public void quadSNP() throws Exception {
         DrawableNode seg4 = sg.getNodes().get(4);
         DrawableNode seg5 = sg.getNodes().get(5);
 
-        DrawableSNP snp = new DrawableSNP(seg0, seg5, sg.getChildren(seg0).stream()
-                                                                        .map(DrawableSegment.class::cast)
-                                                                        .collect(Collectors.toSet()));
         assertNotNull(seg0.createSNPIfPossible(sg));
```

Comment:
lines 111-113 broke the test because creating an SNP adds it to the children of the parent (and the parents of the child), which brings the total number of children to 5 on line 114.
I think that this is very unexpected behaviour, and would very much like to see a better method.
One option would be to use a factory (static) method in DrawableSNP that creates an SNP, and then also updates parents / children / adds SNP to subGraph / removes mutations from subGraph (this method would take the SubGraph to do this).
I don't think this is the best solution, but I can't think of any better ones at this time.